frontend/packages/knative-plugin/src/components/test-function/TestFunctionModal.tsx

```diff
@@ -0,0 +1,76 @@
+import * as React from 'react';
+import { Modal, ModalVariant } from '@patternfly/react-core';
+import { FormikProps, FormikValues } from 'formik';
+import { useTranslation } from 'react-i18next';
+import ServerlessFxIcon from '@console/dev-console/src/components/import/ServerlessFxIcon';
+import {
+  ModalTitle,
+  ModalBody,
+  ModalSubmitFooter,
+  ModalComponentProps,
+} from '@console/internal/components/factory/modal';
+import { useK8sGet } from '@console/internal/components/utils/k8s-get-hook';
+import { ServiceModel } from '../../models';
+import { ServiceKind } from '../../types';
+import ModalBodyWrapper from './modal-components/ModalBodyWrapper';
+import './TestFunctionModal.scss';
+
+interface TestFunctionModalProps {
+  service: ServiceKind;
+}
+
+type Props = FormikProps<FormikValues> & TestFunctionModalProps & ModalComponentProps;
+
+const TestFunctionModal: React.FC<Props> = (props) => {
+  const { t } = useTranslation();
+  const { handleSubmit, cancel, close, isSubmitting, status, service, setFieldValue } = props;
+  const { name, namespace } = service.data.metadata;
+  const [svc, loaded] = useK8sGet<ServiceKind>(ServiceModel, name, namespace);
+
+  React.useEffect(() => {
+    if (loaded) {
+      setFieldValue('endpoint', svc.status.url);
+    }
+  }, [loaded, svc, setFieldValue]);
+
+  return (
+    <>
+      <Modal
+        isOpen
+        showClose={false}
+        variant={ModalVariant.large}
+        position="top"
+        positionOffset="3%"
+        className="kn-test-sf-modal__size"
+      >
+        <form className="modal-content" onSubmit={handleSubmit}>
+          <ModalTitle close={close}>
+            <span className="kn-test-sf-modal__title">
+              <ServerlessFxIcon />
+            </span>
+            {t(`knative-plugin~Test Serverless Function`)}
```

Comment:
Can we show the function name here as well? Maybe with a KSVC badge? Similar to the topology sidebar, for example:

![image](https://user-images.githubusercontent.com/139310/229061638-10b99575-40be-4f76-b24e-8af875e697f6.png)

But I would not add a link here, if the badge always includes a link the function name alone is fine.

### Test Serverless Function **(KSVC)** kn-func-quarkus-http